nemo_curator/stages/text/deduplication/semantic.py

```diff
@@ -0,0 +1,524 @@
+from nemo_curator.stages.text.io.reader import JsonlReader, ParquetReader
+from nemo_curator.stages.text.io.writer import ParquetWriter
+from nemo_curator.utils.file_utils import create_or_overwrite_dir
+
+
+@dataclass
+class TextSemanticDeduplicationWorkflow:
+    """
+    Monolithic workflow for end-to-end text semantic deduplication.
+
+    This workflow combines:
+    1. Text embedding generation (configurable executor)
+    2. Semantic deduplication (configurable executor for pairwise stage)
+    3. Duplicate removal (configurable executor)
+
+    Supports flexible executor configuration - can use a single executor for all stages
+    or different executors for different phases.
+    """
+
+    # Input/Output configuration
+    input_path: str | list[str]
+    output_path: str
+    cache_path: str | None = None
+    perform_removal: bool = True
+    # Embedding generation parameters
+    text_field: str = "text"
+    embedding_field: str = "embeddings"
+    model_identifier: str = "sentence-transformers/all-MiniLM-L6-v2"
+    embedding_max_seq_length: int = 512
+    embedding_max_chars: int | None = None
+    embedding_padding_side: Literal["left", "right"] = "right"
+    embedding_pooling: Literal["mean_pooling", "last_token"] = "mean_pooling"
+    embedding_model_inference_batch_size: int = 256
+    hf_token: str | None = None
+    # Semantic deduplication parameters
+    n_clusters: int = 100
+    id_field: str = CURATOR_DEDUP_ID_STR
+    embedding_dim: int | None = None
+    metadata_fields: list[str] | None = None
+    distance_metric: Literal["cosine", "l2"] = "cosine"
+    which_to_keep: Literal["hard", "easy", "random"] = "hard"
+    eps: float | None = 0.01
+    # K-means clustering parameters
+    kmeans_max_iter: int = 300
+    kmeans_tol: float = 1e-4
+    kmeans_random_state: int = 42
+    kmeans_init: str = "k-means||"
+    kmeans_n_init: int | Literal["auto"] = 1
+    kmeans_oversampling_factor: float = 2.0
+    kmeans_max_samples_per_batch: int = 1 << 15  # 32768
+    # Pairwise similarity parameters
+    ranking_strategy: RankingStrategy | None = None
+    pairwise_batch_size: int = 1024
+    _duplicates_num_row_groups_hint: int | None = None
+    # ID generator parameters
+    use_id_generator: bool = False
+    id_generator_state_file: str | None = None
+    # I/O parameters
+    input_filetype: Literal["jsonl", "parquet"] = "parquet"
+    input_file_extensions: list[str] | None = None
+    input_files_per_partition: int | None = None
+    input_blocksize: int | None = None
+    output_filetype: Literal["jsonl", "parquet"] = "parquet"
+    output_file_extension: str | None = None
+    output_fields: list[str] | None = None
+    read_kwargs: dict[str, Any] = field(default_factory=dict)
+    cache_kwargs: dict[str, Any] = field(default_factory=dict)
+    write_kwargs: dict[str, Any] = field(default_factory=dict)
+    # Execution parameters
+    verbose: bool = True
+    clear_output: bool = True
+    """
```

Comment:
I noticed that these are parameters that are not in the docstring:
```
id_generator_state_file: str | None = None
output_file_extension: str | None = None
output_fields: list[str] | None = None
cache_kwargs: dict[str, Any] = field(default_factory=dict)
```
I noticed that this is in the docstring but not parameters:
```
executor: Single executor for all stages or tuple of (embedding, pairwise, removal) executors
```

Reply:
Thanks for catching that!